### PR TITLE
[script.module.phate89] 1.0.0

### PR DESCRIPTION
**Pull request by phate89**
### Description
It's a helper module for my addons with shared code

### Checklist:
- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [sciprt.foo.bar] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: script.module.phate89/lib/phate89lib/kodiutils.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+def getSetting(setting):
+    return ADDON.getSetting(setting).strip()
+
+def getSettingAsBool(setting):
+    return getSetting(setting).lower() == "true"
+
+def getSettingAsNum(setting):
+    num=0
+    try:
+        num=float(getSetting('lastupdated'))
```
Comment (phil65) [inline script.module.phate89/lib/phate89lib/kodiutils.py]: probably getSetting(setting) ?

File: script.module.phate89/lib/phate89lib/rutils.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            exts = ['.srt', '.sub', '.txt', '.smi', '.ssa', '.ass']
+            subs = filter(lambda x: os.path.splitext(x)[1] in exts, zf.namelist())
+            if index>=len(subs):
+                index=0
+            binData=zf.read(subs[index])
+            zf.close()
+            outName=os.path.join(dataPath,subs[index])
+            
+            if binData:       
+                fp=open(outName,'wb')
```
Comment (phil65) [inline script.module.phate89/lib/phate89lib/rutils.py]: try-except or with-statement for IO, please.

**Comment by phil65** [comment]
One last thing: Addon.xml states GNU GENERAL PUBLIC LICENSE Version 2.1, License text says version 2.

**Comment by phate89** [comment]
ops c/p error. fixed this too. thanks @phil65 !

**Comment by MartijnKaijser** [comment]
I'm gonna sya it again. I really HATE the fact that every one creates their own modules for their own goal which in the end will just die out

**Pull request merged**